From 456828db1911227f6230d0d74ec1a535d629f7da Mon Sep 17 00:00:00 2001
From: Bai Ping <ping.bai@nxp.com>
Date: Thu, 22 Jun 2017 14:55:58 +0800
Subject: [PATCH 1992/5242] MLK-15137-02 driver: clk: Remove new_div_ack check
 in frac pll

commit  c631249216bfbafeb9280e14425c2600a74508e0 from
https://source.codeaurora.org/external/imx/linux-imx.git

If the frac pll is powrer down or hold in reset, new_div_ack
will not be assert. Waiting for ack will failed.

Signed-off-by: Bai Ping <ping.bai@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/clk/imx/clk-frac-pll.c |    6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/drivers/clk/imx/clk-frac-pll.c b/drivers/clk/imx/clk-frac-pll.c
index 0711e1c..ba64c9e 100644
--- a/drivers/clk/imx/clk-frac-pll.c
+++ b/drivers/clk/imx/clk-frac-pll.c
@@ -162,16 +162,12 @@ static int clk_pll_set_rate(struct clk_hw *hw, unsigned long rate,
 	val |= PLL_NEWDIV_VAL;
 	writel_relaxed(val, pll->base + PLL_CFG0);
 
-	/* wait for the new divfi and divff reload successfully */
-	while (!(readl_relaxed(pll->base + PLL_CFG0) & PLL_NEWDIV_ACK))
-		;
 	/* clear the NEV_DIV_VAL */
 	val = readl_relaxed(pll->base + PLL_CFG0);
 	val &= ~PLL_NEWDIV_VAL;
 	writel_relaxed(val, pll->base + PLL_CFG0);
 
-	/* wait for PLL is locked */
-	return clk_wait_lock(pll);
+	return 0;
 }
 
 static const struct clk_ops clk_frac_pll_ops = {
-- 
1.7.9.5

